src/components/expenses/grid/DetailsModal.tsx

```diff
@@ -0,0 +1,36 @@
+import { UseQueryResult } from 'react-query'
+import { observer } from 'mobx-react'
+import { useTranslation } from 'next-i18next'
+
+import { ExpenseResponse } from 'gql/expenses'
+import { useViewExpense } from 'common/hooks/expenses'
+import { usePersonList } from 'common/hooks/person'
+import { ModalStore } from 'stores/dashboard/ModalStore'
+import DetailsDialog from 'components/admin/DetailsDialog'
+
+export default observer(function DetailsModal() {
+  const { selectedRecord } = ModalStore
+  const { data }: UseQueryResult<ExpenseResponse> = useViewExpense(selectedRecord.id)
+  const { data: personList } = usePersonList()
+  const approvedBy = personList?.find((person) => person.id == data?.approvedById)
+
+  const { t } = useTranslation('expenses')
+
+  const dataConverted = [
+    { name: 'ID', value: `${data?.id}` },
+    { name: t('fields.type'), value: `${data?.type}` },
+    { name: t('fields.status'), value: `${data?.status}` },
+    { name: t('fields.currency'), value: `${data?.currency}` },
+    { name: t('fields.amount'), value: `${data?.amount}` },
+    { name: t('fields.vaultId'), value: `${data?.vaultId}` },
+    { name: t('fields.deleted'), value: `${data?.deleted}` },
+    { name: t('fields.description'), value: `${data?.description}` },
+    { name: t('fields.documentId'), value: `${data?.documentId}` },
```

Comment:
Why do you convert all fields to string? This actions seems to be redundant and will lead to `undefined` strings shown to the end user

Reply:
It's because of how `<DetailsDialog/>` component expects data.
```javascript
export type Field = {
  name: string
  value: string | number | JSX.Element
}
```

If I modify it to (_by adding undefined and null for value type_)
```javascript
export type Field = {
  name: string
  value: string | number | JSX.Element | undefined | null
}
```
then this will work:
```javascript
const dataConverted = [
    { name: 'ID', value: `${data?.id}` },
    { name: t('fields.type'), value: data?.type },
    { name: t('fields.status'), value: data?.status },
    { name: t('fields.currency'), value: data?.currency },
    { name: t('fields.amount'), value: data?.amount },
    { name: t('fields.vaultId'), value: data?.vaultId },
    { name: t('fields.deleted'), value: data?.deleted?.toString() },
    { name: t('fields.description'), value: data?.description },
    { name: t('fields.documentId'), value: data?.documentId },
    {
      name: t('fields.approvedById'),
      value: approvedBy ? `${approvedBy.firstName} ${approvedBy.lastName} (${approvedBy.id})` : '',
    },
  ]

  return <DetailsDialog data={dataConverted} />
```

Null is required because **documentId** can be null.
Undefined, obviously because the rest of the properties of ExpenseResponse can be undefined.
Deleted property is of type Boolean | undefined so it need to be converted to string if set.

**TL;DR**:
I agree. I want to remove the redundant conversation to strings.
But to do so I need to change the possible types of value property in type Field (_DetailsDialog.tsx_) so Typescript can be happy.

Reply:
It's best to define the `value` as `value: React.ReactNode` which includes all of the mentioned types